gtda/diagrams/_utils.py

```diff
@@ -10,8 +10,10 @@ def _subdiagrams(X, homology_dimensions, remove_dim=False):
     the same number of points in each homology dimension."""
     n = len(X)
     if len(homology_dimensions) == 1:
+        # Reshape ensures copy
```

Comment:
Let me present what i found.
Reading the documentation (see [the Returns section](https://numpy.org/doc/stable/reference/generated/numpy.reshape.html)), it seems that it returns a copy only if it cannot reshape the diagram. So, for example, if `all(X[:, :, 2] == homology_dimensions[0])`, then the shape would not be changed and the array might not be copied.

Reply:
I tested precisely your scenario and found that this was not the case. The reason is that `X[X[:, :, 2] == dim]` can never be of the same shape as `X`. Could you double check?

Reply:
Let me also say that it is *essential* that `_subdiagrams` always returns copies, as we do all sorts of in-place things to subdiagrams in several transformers.

Reply:
True, my bad! I forgot what the output of `X[X[:, :, 2] == dim]` looks like.